app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartController.scala

```diff
@@ -76,9 +77,26 @@ class StartController @Inject()(
 
             for {
               record <- subscriptionJourneyService.getMandatoryJourneyRecord(ContinueId(continueId))
-              _ <- subscriptionJourneyService.saveJourneyRecord(
-                    record.copy(cleanCredsAuthProviderId = Some(agent.authProviderId)))
-            } yield Redirect(routes.TaskListController.showTaskList())
+              subscriptionStatus <- subscriptionService
```

Comment:
I wonder if we should move this whole block to the subscriptionService under a method like redirectAccordingToStatus(...) or something?  Be nice to make the controllers less complex ...

Reply:
👍

Reply:
Actually, the logic looks similar to handlePartiallySubscribedAndRedirect(...), no?   These two blocks do similar things, not sure how to consolidate, or even if it's worth it... thoughts?

Reply:
umm i've moved it into the service. I think those 2 methods are different enough to not warrant consolidating them, there will be too many variables.